fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java

```diff
@@ -561,6 +562,12 @@ protected void runRunningJob() throws AlterCancelException {
 
     private void onFinished(OlapTable tbl) {
         TabletInvertedIndex invertedIndex = GlobalStateMgr.getCurrentInvertedIndex();
+        // 
+        // partition visible version won't update in schema change, so we need make global
+        // dictionary invalid after schema change.
+        for (Column column : tbl.getColumns()) {
+            IDictManager.getInstance().removeGlobalDict(tbl.getId(), column.getName());
```

Comment:
you should add some check. such as only for string columns. removeGlobalDict need to synchronous

Reply:
There are no any synchronized in removeGlobalDict in fact.
synchronous means create a cache view
```
    public LoadingCache<K, V> synchronous() {
        return this.cacheView == null ? (this.cacheView = new LocalAsyncLoadingCache.LoadingCacheView(this)) : this.cacheView;
    }
```

invalidate means remove the key for the cache
```
        public void invalidate(Object key) {
            this.asyncCache().cache().remove(key);
        }
```
so there are no any performance issue here.